### ISSUE TITLE
Some bots keep fighting too low-level mobs

### ISSUE THREAD

**Issue by Dreathean** [liyunfan1223/mod-playerbots#12]
This may be a case of me figuring how to update their interval to teleport to +/- 3 level location, but I've noticed at least in the blood elf starting zone that some (not all) keep fighting mobs too low for their level. For example this paladin lvl 8 fighting lvl 1 mob in the same spot for hours.

Thank you for everything!

![WoWScrnShot_073023_144723](https://github.com/liyunfan1223/mod-playerbots/assets/132776330/2a27a112-63ec-4af9-af06-2fe76161ad19)
![WoWScrnShot_073023_144737](https://github.com/liyunfan1223/mod-playerbots/assets/132776330/ef8f9ba6-3d94-40f2-ae8b-fbb089af151a)

**Comment by liyunfan1223** [comment]
Thank you, this is a good issue!

I temporarily solved this issue by modifying the way the random bots teleport. Now the random bots will teleport to the mobs of the corresponding level more accurately, so that they can always fight with the mobs of the corresponding level.

https://github.com/liyunfan1223/mod-playerbots/commit/0969db4c8998cd862bcd0e13b5de5d192bccc885

Also, this question touches on how random bots behave. I don't think it's the best way to teleport random bots directly to mobs, maybe they should teleport to the main city first, and then move to an area of ​​appropriate level for leveling from the main city, but this obviously requires more efforts to develop. So I solved the issue in this way for the time being.

**Comment by garfieldz90** [comment]
> 

I agree, is not good directly to teleport bots to mobs zone, but they just stay in main town walk around don’t go outside, especially in huge town, but don’t focus on this right now cause this need many many time! fix other bugs better.

**Comment by Dreathean** [comment]
Awesome, thank you for the quick fix! I agree, it does sound very time consuming and challenging to re-work the teleporting in that way. I wonder if would be possible (in the future perhaps) to run a "check" when or prior to teleporting to see if there are multiple mobs in aggro range? To prevent the issue of a playerbot teleporting in the middle of multiple mobs and getting wrecked, which at this time leads to the bot repeatedly corpse-running to their death over and over until they teleport to another area. Just a thought, no idea how difficult that would be to code.

**Comment by liyunfan1223** [comment]
> Awesome, thank you for the quick fix! I agree, it does sound very time consuming and challenging to re-work the teleporting in that way. I wonder if would be possible (in the future perhaps) to run a "check" when or prior to teleporting to see if there are multiple mobs in aggro range? To prevent the issue of a playerbot teleporting in the middle of multiple mobs and getting wrecked, which at this time leads to the bot repeatedly corpse-running to their death over and over until they teleport to another area. Just a thought, no idea how difficult that would be to code.

Thanks! I have also noticed this problem. Currently they will teleport to another position when they dead too much times (currently 5 times), which help them find a better place for leveling. This solve the problem to some extent.

**Comment by liyunfan1223** [comment]
I'd like to close this question because it's temporarily resolved, and we can continue the discussion elsewhere about how to make random bots behave more like real player, how to keep them alive as much as possible while leveling, etc. Thank you each and every one of you for participating.

**Comment by Dreathean** [comment]
> > Awesome, thank you for the quick fix! I agree, it does sound very time consuming and challenging to re-work the teleporting in that way. I wonder if would be possible (in the future perhaps) to run a "check" when or prior to teleporting to see if there are multiple mobs in aggro range? To prevent the issue of a playerbot teleporting in the middle of multiple mobs and getting wrecked, which at this time leads to the bot repeatedly corpse-running to their death over and over until they teleport to another area. Just a thought, no idea how difficult that would be to code.
> 
> Thanks! I have also noticed this problem. Currently they will teleport to another position when they dead too much times (currently 5 times), which help them find a better place for leveling. This solve the problem to some extent.

Excellent idea, that is a great solution for now! Thank you!
